tutorial_code/example2_2_frustrated_loop_anneal.py

```diff
@@ -203,87 +227,126 @@ def run_experiment(param, shim, stats, embeddings, alpha_Phi=0., alpha_J=0.):
+        use_cache (bool): When True an attempt is made to load (save) data from
+            (to) the directory cached_experimental_data.
     """
-
-    prefix = f'example2_2_aPhi{alpha_Phi}_aJ{alpha_J}'
-
-    data_dict = {'param': param, 'shim': shim, 'stats': stats}
-    data_dict = load_experiment_data(prefix, data_dict)
-
+    if use_cache:
+        solver_name = param["sampler"].properties["chip_id"]
+        prefix = f"{solver_name}_example2_2_aPhi{alpha_Phi}_aJ{alpha_J}"
+        data_dict = {"param": param, "shim": shim, "stats": stats}
+        data_dict = load_experiment_data(prefix, data_dict)
+    else:
+        data_dict = None
     if data_dict is not None:
-        param = data_dict['param']
-        shim = data_dict['shim']
-        stats = data_dict['stats']
+        param = data_dict["param"]
+        shim = data_dict["shim"]
+        stats = data_dict["stats"]
 
     else:
-        for iteration in tqdm(range(param['num_iters']), total=param['num_iters']):
-            if iteration < 100:
-                shim['alpha_Phi'] = 0.
+        # prev_execution_time = 441.86028
+        print("Collection of data typically requires several minutes")
+        for iteration in tqdm(range(param["num_iters"]), total=param["num_iters"]):
+            if iteration < param["num_iters_unshimmed_flux"]:
+                shim["alpha_Phi"] = 0.0
             else:
-                shim['alpha_Phi'] = alpha_Phi
-            if iteration < 200:
-                shim['alpha_J'] = 0.
+                shim["alpha_Phi"] = alpha_Phi
+            if iteration < param["num_iters_unshimmed_J"]:
+                shim["alpha_J"] = 0.0
             else:
-                shim['alpha_J'] = alpha_J
+                shim["alpha_J"] = alpha_J
             run_iteration(param, shim, embeddings, stats)
+        if use_cache:
+            save_experiment_data(prefix, {"param": param, "shim": shim, "stats": stats})
+
+    paper_plots_example2_2(
+        nominal_couplings=shim["nominal_couplings"],
+        all_fbos=stats["all_fbos"],
+        all_couplings=stats["all_couplings"],
+        mags=stats["mags"],
+        frust=stats["frust"],
+    )
 
-        save_experiment_data(
-            prefix,
-            {'param': param, 'shim': shim, 'stats': stats}
-        )
 
-    plot_data(all_fbos=stats['all_fbos'], mags=stats['mags'],
-              all_couplings=stats['all_couplings'], frust=stats['frust'],
-              all_alpha_phi=stats['all_alpha_Phi'], all_alpha_j=stats["all_alpha_J"],
-              coupler_orbits=shim['coupler_orbits'], alpha_phi=shim['alpha_Phi'], alpha_j=shim['alpha_J'],
-              coupling=param["coupling"], L=param["L"])
-    paper_plots_example2_2(nominal_couplings=shim['nominal_couplings'],
-                           all_fbos=stats['all_fbos'],
-                           all_couplings=stats['all_couplings'],
-                           mags=stats['mags'],
-                           frust=stats['frust'])
+def main(
```

Comment:
add 'max_num_emb' as an argument here defaulted to float('Inf')
Document it.
Pass it to embed_loops()